src/Tools/ExternalAccess/FSharp/Navigation/FSharpDocumentNavigationService.cs

```diff
@@ -13,88 +13,77 @@
-
-        [Obsolete("Call overload that takes a CancellationToken", error: false)]
-        public bool CanNavigateToSpan(Workspace workspace, DocumentId documentId, TextSpan textSpan)
-            => CanNavigateToSpan(workspace, documentId, textSpan, CancellationToken.None);
+[ExportWorkspaceService(typeof(IFSharpDocumentNavigationService)), Shared]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal class FSharpDocumentNavigationService(IThreadingContext threadingContext)
+    : IFSharpDocumentNavigationService
+{
+    [Obsolete("Call overload that takes a CancellationToken", error: false)]
+    public bool CanNavigateToSpan(Workspace workspace, DocumentId documentId, TextSpan textSpan)
+        => CanNavigateToSpan(workspace, documentId, textSpan, CancellationToken.None);
 
-        public bool CanNavigateToSpan(Workspace workspace, DocumentId documentId, TextSpan textSpan, CancellationToken cancellationToken)
-        {
-            var service = workspace.Services.GetService<IDocumentNavigationService>();
-            return _threadingContext.JoinableTaskFactory.Run(() =>
-                service.CanNavigateToSpanAsync(workspace, documentId, textSpan, cancellationToken));
-        }
+    public bool CanNavigateToSpan(Workspace workspace, DocumentId documentId, TextSpan textSpan, CancellationToken cancellationToken)
+    {
+        var service = workspace.Services.GetService<IDocumentNavigationService>();
+        return threadingContext.JoinableTaskFactory.Run(() =>
+            service.CanNavigateToSpanAsync(workspace, documentId, textSpan, cancellationToken));
+    }
 
-        [Obsolete("Call overload that takes a CancellationToken", error: false)]
-        public bool CanNavigateToLineAndOffset(Workspace workspace, DocumentId documentId, int lineNumber, int offset)
-            => CanNavigateToLineAndOffset(workspace, documentId, lineNumber, offset, CancellationToken.None);
+    [Obsolete("Call overload that takes a CancellationToken", error: false)]
+    public bool CanNavigateToLineAndOffset(Workspace workspace, DocumentId documentId, int lineNumber, int offset)
+        => CanNavigateToLineAndOffset(workspace, documentId, lineNumber, offset, CancellationToken.None);
 
-        public bool CanNavigateToLineAndOffset(Workspace workspace, DocumentId documentId, int lineNumber, int offset, CancellationToken cancellationToken)
-        {
-            var service = workspace.Services.GetService<IDocumentNavigationService>();
-            return _threadingContext.JoinableTaskFactory.Run(() =>
-                service.CanNavigateToLineAndOffsetAsync(workspace, documentId, lineNumber, offset, cancellationToken));
-        }
+    [Obsolete("Call overloads that take a span or position", error: false)]
+    public bool CanNavigateToLineAndOffset(Workspace workspace, DocumentId documentId, int lineNumber, int offset, CancellationToken cancellationToken)
+        => false;
```

Comment:
Would this regress F#? Are we planning a dual insertion with them? If not, might need to keep the implementation here until they moved off this API

Reply:
I've checked F# and they do not use this.  But i'll double check with them again on this :)